microservices/webapp/vite-react/src/routes/Rooms.tsx

```diff
@@ -0,0 +1,125 @@
+import { NativeSelect, Table } from "@mantine/core"
+import { useState } from 'react';
+import { MultiSelect } from '@mantine/core';
+import { TextInput } from '@mantine/core';
+
+export function Rooms() {
+
+  const [featureValue, setFeatureValue] = useState('name');
+  const [textValue, setTextValue] = useState('');
+
+  const rooms = [
+    {
+      name: "Room 2312",
+      location: "A-Wing 2nd Floor",
+      description: "This is an awesome room filled with so much fun!",
+      type: "Fun Room",
+      features: "Friendly, Awesome whatever",
+      id: 1,
+      created_at: "Jan 12, 2012",
+      last_updated_at: "Jan 12, 2015"
+    },
+    {
+      name: "Room 1001",
+      location: "B-Wing 1st Floor",
+      description: "This room is great for studying or working alone.",
+      type: "Study Room",
+      features: "Quiet, Has a desk and chair",
+      id: 2,
+      created_at: "Feb 15, 2013",
+      last_updated_at: "Feb 18, 2015"
+    },
+    {
+      name: "Room 2020",
+      location: "C-Wing 2nd Floor",
+      description: "This is a cozy room with a great view of the city.",
+      type: "Comfort Room",
+      features: "Warm lighting, Comfortable furniture",
+      id: 3,
+      created_at: "Mar 22, 2014",
+      last_updated_at: "Mar 24, 2016"
+    },
+    {
+      name: "Room 305",
+      location: "D-Wing 3rd Floor",
+      description: "This is a large conference room that can fit up to 20 people.",
+      type: "Conference Room",
+      features: "Projector, Whiteboard, Conference table and chairs",
+      id: 4,
+      created_at: "Apr 18, 2015",
+      last_updated_at: "Apr 20, 2018"
+    },
+    {
+      name: "Room 405",
+      location: "E-Wing 4th Floor",
+      description: "This is a bright and airy room that's perfect for yoga or meditation.",
+      type: "Fitness Room",
+      features: "Large windows, Yoga mats, Meditation cushions",
+      id: 5,
+      created_at: "May 25, 2016",
+      last_updated_at: "May 28, 2019"
+    }
+  ];
+
+  var rows = rooms.map((room) => {
+    if (room.hasOwnProperty(featureValue)) {
+      const value = room[featureValue];
```

Comment:
```
src/routes/Rooms.tsx(66,21): error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ name: string; location: string; description: string; type: string; features: string; id: number; created_at: string; last_updated_at: string; }'.
```

Reply:
I'm not too sure what is causing this error, but it does have to do with the room search. I am thinking that a search functionality that runs entirely client-side is not efficient (since the client would have to pull all room data). Since this kind of functionality is also what's causing the frontend build to fail, what if we **scrap this and use a GET request to filter**? This would be parameters in the query string that the backend can do to perform searching for us.

Here is what the API spec would look like:
```
/items:
    get:
      summary: Gets a list of items. For now, the only kind of item is a room.
      parameters:
        - name: page
          in: query
          description: Page number
          required: false
          schema:
            type: integer
            format: int64
            minimum: 1
            default: 1
        - name: per_page
          in: query
          description: Number of items per page
          required: false
          schema:
            type: integer
            format: int64
            minimum: 1
            maximum: 100
            default: 10
        - name: nameSearch
          in: query
          description: Search for a room by name
          required: false
          schema:
            type: string
        - name: locationSearch
          in: query
          description: Search for a room by location
          required: false
          schema:
            type: string
        - name: descriptionSearch
          in: query
          description: Search for a room by description
          required: false
          schema:
            type: string
```

An as an example for queries:
* `... /items?nameSearch=foo&descriptionSearch=bar&page=1`

As a reminder, none of us need to worry about constructing query strings because we will have a client library that handles all requests for us.